src/vs/workbench/contrib/terminal/browser/widgets/widgetManager.ts

```diff
@@ -4,15 +4,21 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { IDisposable } from 'vs/base/common/lifecycle';
+import { ILayoutService } from 'vs/platform/layout/browser/layoutService';
 import { ITerminalWidget } from 'vs/workbench/contrib/terminal/browser/widgets/widgets';
 
 export class TerminalWidgetManager implements IDisposable {
 	private _container: HTMLElement | undefined;
 	private _attached: Map<string, ITerminalWidget> = new Map();
 
+	constructor(
+		@ILayoutService private readonly _layoutService: ILayoutService,
+	) {
+	}
+
 	attachToElement(terminalWrapper: HTMLElement) {
 		if (!this._container) {
-			this._container = document.createElement('div');
+			this._container = this._layoutService.container.ownerDocument.createElement('div');
```

Comment:
You can still use `document.createElement`, in fact we are currently blocking `createElement` on floating windows because this breaks `instanceof HTMLElement` checks. We might revisit this decision later, but for now I suggest to go back to `document.createElement`